CMakeLists.txt

```diff
@@ -20,70 +20,95 @@ set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
 # Search for dependencies. #
 ############################
 
-set(CATKIN_PKGS ${CATKIN_PKGS}
-    # MPI-IS dependencies
-    mpi_cmake_modules
-    pybind11_catkin
-)
-find_package(catkin REQUIRED COMPONENTS ${CATKIN_PKGS})
-
-# catkin customised search macros.
-SET(BOOST_COMPONENTS
-      filesystem system thread program_options unit_test_framework python)
-search_for_python()
-search_for_boost()
-search_for_eigen()
-catkin_add_required_dependency("pinocchio >= 2.0")
-
-######################################################
-# define the include directory of all ${CATKIN_PKGS} #
-######################################################
-include_directories(
-  ${PROJECT_SOURCE_DIR}/include
-  ${catkin_INCLUDE_DIRS}
-)
+#
+# Dependencies
+#
+find_package(pybind11 CONFIG REQUIRED)
+find_package(Eigen3 REQUIRED)
+find_package(pinocchio REQUIRED)
+find_package(eiquadprog REQUIRED)
+SET(BOOST_COMPONENTS system python)
+find_package(Boost REQUIRED)
```

Comment:
find_package(Boost REQUIRED COMPONENT python)

Reply:
I tried to change to `find_package(Boost REQUIRED COMPONENT python)` based on your suggestion, but then I am facing this error:

```
--- stderr: blmc_controllers
CMake Error at /usr/local/Cellar/cmake/3.19.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find Boost (missing: python) (found version "1.74.0")
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.19.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:577 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.19.1/share/cmake/Modules/FindBoost.cmake:2193 (find_package_handle_standard_args)
  CMakeLists.txt:29 (find_package)
```

Any thoughts?

Reply:
`find_package(Boost REQUIRED COMPONENTS thread)` in CMakeLists.txt
and
`find_dependency(Boost REQUIRED COMPONENTS thread)` in Config.cmake.in

This works for the `solver` package.

Reply:
Could it be that you are missing the `S` in `COMPONENTS`?

Reply:
You are right, the `S` was missing before.

According to `FindBoost` [1] page, it is required to specify the version of python to require (`python27` or `python36`)

When I try with `find_package(Boost REQUIRED COMPONENT python36)` I am getting:

```
CMake Error at /usr/local/Cellar/cmake/3.19.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find Boost (missing: python36) (found version "1.74.0")
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.19.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:577 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.19.1/share/cmake/Modules/FindBoost.cmake:2193 (find_package_handle_standard_args)
  CMakeLists.txt:29 (find_package)
```

When I try with `find_package(Boost REQUIRED COMPONENT python39)`, I am not getting an error on my machine anymore. So it looks like one has to specify the python version used by the system (on my mac python 3.9 is the default).

Do you have an idea how to figure out the current python version and then use it in the find boost command?

[1]: https://cmake.org/cmake/help/latest/module/FindBoost.html

Reply:
have you tried the last comment of this answer: https://stackoverflow.com/questions/56067518/cmake-could-not-find-boost-python

Reply:
If this works we can create a `get_python_version(python_version)` in mpi_cmake_module
Or just add these lines for all packages using boost python...

Reply:
Thanks for the link. This seems to work.

I've added the lines in the script directly as I assume there won't be much boost-python usage from our code.

I've added similar lines in the `Config.cmake.in` file. Is that okay to call `find_package(PythonLibs REQUIRED)` from there?

Reply:
Yes it is ok I think...
Did you try to depend on this package?

Reply:
I added a dependency from the momentumopt package to this package. This seems to work (it compiles fine).